### PR TITLE
Apidoc type consistency

### PR DESCRIPTION
**Pull request by notbakaneko**
- `number` is now either `integer` or `float` (a reversal from https://github.com/ppy/osu-web/pull/10618#discussion_r1358134093 which is what made me think about this); while JSON doesn't define either, JSON Schema has at least `integer` and it's more useful for API consumers in other languages to know if a number if an integer or not.
- date strings should now all be `Timestamp`
- `GameMode` is now `Ruleset` to match our naming for it going forward

There's other stuff like more linking to types and the `Multiplayer*` structures being in code blocks for some reason❔ that will be done separately

... it would be nice if we could just stick attributes on the transformers or something and have the correct type and markdown tables generated 👀

### CODE REVIEW

File: resources/views/docs/_structures/forum_post.md [hunk 1]
```diff
@@ -5,11 +5,11 @@ Field        | Type                     | Description
...
-edited_by_id | number?                  | |
-forum_id     | number                   | |
-id           | number                   | |
-topic_id     | number                   | |
-user_id      | number                   | |
+edited_by_id | ingeter?                 | |
+forum_id     | ingeter                  | |
+id           | ingeter                  | |
+topic_id     | ingeter                  | |
+user_id      | ingeter                  | |
```
Comment (Swekka) [inline resources/views/docs/_structures/forum_post.md:12]: typo

File: app/Http/Controllers/NewsController.php [hunk 2]
```diff
@@ -26,10 +26,10 @@ class NewsController extends Controller
      * ------------------------- | ----------------------------- | -----
      * cursor_string             | [CursorString](#cursorstring) | |
      * news_posts                | [NewsPost](#newspost)[]       | Includes `preview`.
-     * news_sidebar.current_year | number                        | Year of the first post's publish time, or current year if no posts returned.
+     * news_sidebar.current_year | integer                       | Year of the first post's publish time, or current year if no posts returned.
      * news_sidebar.news_posts   | [NewsPost](#newspost)[]       | All posts published during `current_year`.
      * news_sidebar.years        | number[]                      | All years during which posts have been published.
```
Comment (Swekka) [inline app/Http/Controllers/NewsController.php:31]:
```suggestion
     * news_sidebar.years        | integer[]                      | All years during which posts have been published.
```

File: app/Http/Controllers/HomeController.php [hunk 3]
```diff
@@ -166,10 +166,10 @@ public function quickSearch()
...
-     * Field | Type   | Description
-     * ----- | ------ | -----------
-     * data  | T[]    | |
-     * total | number | |
+     * Field | Type    | Description
+     * ----- | ------- | -----------
+     * data  | T[]     | |
+     * total | integer | |
      *
      * @queryParam mode Either `all`, `user`, or `wiki_page`. Default is `all`. Example: all
```
Comment (Swekka) [inline app/Http/Controllers/HomeController.php:174]:
```suggestion
     * @queryParam mode string Either `all`, `user`, or `wiki_page`. Default is `all`. Example: all
```
Without this it defaults to integer

File: app/Http/Controllers/RankingController.php [hunk 4]
```diff
@@ -111,7 +111,7 @@ public function __construct()
      *
      * Returns [Rankings](#rankings)
      *
-     * @urlParam mode string required [GameMode](#gamemode). Example: mania
+     * @urlParam mode string required [Ruleset](#ruleset). Example: mania
      * @urlParam type string required [RankingType](#rankingtype). Example: performance
      *
      * @queryParam country Filter ranking by country code. Only available for `type` of `performance`. Example: JP
```
Comment (Swekka) [inline app/Http/Controllers/RankingController.php:117]:
```suggestion
     * @queryParam country string Filter ranking by country code. Only available for `type` of `performance`. Example: JP
```
Same here

File: resources/views/docs/_structures/multiplayer_score.md [hunk 5]
```diff
@@ -4,18 +4,18 @@ Score data.
...
 `rank`             | `rank`                     |  |
-`total_score`      | `number`                   |  |
-`accuracy`         | `number`                   |  |
-`max_combo`        | `number`                   |  |
+`total_score`      | `integer`                  |  |
+`accuracy`         | `float`                    |  |
+`max_combo`        | `integer`                  |  |
 `mods`             | `Mod[]`                    |  |
 `statistics`       | `Statistics`               |  |
 `passed`           | `bool`                     |  |
```
Comment (Swekka) [inline resources/views/docs/_structures/multiplayer_score.md:18]:
```suggestion
`passed`           | `boolean`                  |  |
```

**Review by Swekka — APPROVED** [review]

**Review by nanaya — APPROVED** [review]

**Pull request merged**